karabo_data/geometry2.py

```diff
@@ -62,14 +62,23 @@ def centre(self):
             + (0.5 * self.fs_vec * self.fs_pixels)
         )
 
-    def to_crystfel_geom(self, p, a):
+    def to_crystfel_geom(self, p, a, ss_dims, fs_dims, extra_dim):
         name = 'p{}a{}'.format(p, a)
         c = self.corner_pos
+        dims = [p, 'ss', 'fs']
+        if extra_dim:
+            dims.insert(extra_dim[0]-1, extra_dim[1])
+        dims_str = ''
+        for num, value in enumerate(dims):
+            dims_str += '{}/dim{} = {}\n'.format(name, num+1, value)
```

Comment:
I think this will also go slightly wrong,  because it will start at `dim0` instead of `dim1`. As I mentioned in another comment, I think we should leave this particular bit of flexibility for a separate PR.

Reply:
I should start with dim1 as I add num+1.

Reply:
Yes, my mistake!